### PR TITLE
added custom tag

### PR DESCRIPTION
**Pull request by LIUSHANYIN**

### CODE REVIEW

File: src/components/Icons/index.tsx [hunk 1]
```diff
@@ -10,21 +10,25 @@ interface IProps {
   color?: string;
   size?: string;
   classPrefix?: string;
-  tag?: string;
+  tag?: 'a';
+  url?: string;
```
Comment (mxdi9i7) [inline src/components/Icons/index.tsx:14]: icons shouldn't have url and replace

File: src/components/Icons/index.tsx [hunk 2]
```diff
@@ -10,21 +10,25 @@ interface IProps {
...
   badge,
   color,
   size,
-  classPrefix = baseClass
+  classPrefix = baseClass,
+  tag,
+  url,
+  replace
 }: IProps) {
+  const CustomTag = tag || 'a';
```
Comment (mxdi9i7) [inline src/components/Icons/index.tsx:31]: custom icon tag should be 'i' by default

File: src/components/Icons/index.tsx [hunk 3]
```diff
@@ -55,12 +59,26 @@ export default function Icon({
...
+    Object.assign(iconProps, {
+      href: url
+    });
+    if (replace) {
+      Object.assign(iconProps, {
+        target: '_self'
+      });
+    } else {
+      Object.assign(iconProps, {
+        target: '_blank'
```
Comment (mxdi9i7) [inline src/components/Icons/index.tsx:72]: these don't belong here

Comment (LIUSHANYIN) [reply src/components/Icons/index.tsx:72]: ok. I just think maybe one day we need the icon to be an 'a' tag. I will delete these codes.

File: src/components/Icons/index.tsx [hunk 4]
```diff
@@ -10,21 +10,21 @@ interface IProps {
   color?: string;
   size?: string;
   classPrefix?: string;
-  tag?: string;
+  tag?: 'i';
```
Comment (z695245771) [inline src/components/Icons/index.tsx:13]: i think tag name should be a string here. 'i' is just the default value, not enum

**Comment by mxdi9i7** [comment]
Missing story

**Pull request merged**